src/core/ledger.js

```diff
@@ -20,6 +19,47 @@ Ledger.from_json = function(v) {
 
 Ledger.space = require('./ledgerspaces');
 
+function hash(hex) {
+  return sha512half(new Buffer(hex, 'hex'));
+}
+
+function hashTransaction(txBlobHex) {
+  const prefix = hashprefixes.HASH_TX_ID.toString(16).toUpperCase();
+  return hash(prefix + txBlobHex);
+}
+
+function padLeftZero(string, length) {
+  return Array(length - string.length + 1).join('0') + string;
```

Comment:
Avoiding unnecessary ES6 for now since this is moving to a new repo soon and it might not have babel setup right away